src/coprocessor/codec/error.rs

```diff
@@ -35,6 +32,21 @@ pub const ERR_DIVISION_BY_ZERO: i32 = 1365;
 pub const ERR_DATA_TOO_LONG: i32 = 1406;
 pub const ERR_DATA_OUT_OF_RANGE: i32 = 1690;
 
+macro_rules! handle_invalid_time_error {
```

Comment:
Hummm. Why is it extracted to a macro? In most cases we would like to avoid the usage of macro if it can be replaced by similar normal codes, because macro is very editor / IDE unfriendly and error message unfriendly.

Reply:
In fact,i don't want to use macro,too. my origin code look like this.
```
    pub fn handle_invalid_time_error<'a, T>(
        ctx: &mut EvalContext,
        err: Error,
    ) -> Result<Option<Cow<'a, T>>>
    where
        T: Clone,
{
    ....
}
```

but when use it in the date_format like this
```
Error::handle_invalid_time_error::<[u8]>(
                ctx,
                Error::incorrect_datetime_value(&format!("Incorrect datetime value: '{}'", t)),
);
```
it will get an error.
```
the trait `std::clone::Clone` is not implemented for `[u8]`
```

Reply:
Hmmm. I think actually you need `T` to be `T: Display` so that it can be printed like this `format!("Incorrect datetime value: '{}'", t)`.

Reply:
The return type of your function might be wrong. How about `handle_invalid_time_error(ctx: ..., err: ...) -> Result<()>`?

Reply:
if change to Result<()>.it will get error.
```
 expected enum `std::option::Option`, found ()
```

if change to Result<Option<()>>..it will still get error.

```
expected enum `std::borrow::Cow`, found ()
```

Reply:
fix it

Reply:
Where did you get the error? I mean, where are these types being used? Seems that there are other places expecting this type, which is not necessary actually.